### PR TITLE
Add clang build to CI.

### PR DESCRIPTION
**Pull request by gengjiawen**
A follow up to https://github.com/jerryscript-project/jerryscript/pull/4684

JerryScript-DCO-1.0-Signed-off-by: Jiawen Geng technicalcute@gmail.com

### CODE REVIEW

File: .github/workflows/gh-actions.yml [hunk 1]
```diff
@@ -129,6 +129,14 @@ jobs:
       - run: $RUNNER -q --unittests
       - run: $RUNNER -q --unittests --build-debug
 
+  Clang_Tests:
```
Comment (rerobika) [inline .github/workflows/gh-actions.yml:132]: `Clang_Unit_Build_Option_Tests:`

**Review by rerobika — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-actions.yml [hunk 2]
```diff
@@ -129,6 +129,14 @@ jobs:
...
+    runs-on: ubuntu-latest
+    env:
+      CC: clang
+    steps:
+      - uses: actions/checkout@v2
+      - run: $RUNNER -q --unittests
```
[the remark this answers is not in the document]
Comment (gengjiawen) [reply .github/workflows/gh-actions.yml:138]: Clang support for lto can't not pass. Is there a way to turn it off.
If not, can I use `python tools/build.py --line-info=ON --logging=ON --error-messages=On --lto=Off` ?

[the remark this answers is not in the document]
Comment (gengjiawen) [reply .github/workflows/gh-actions.yml:138]: Looks ` --buildoptions=--lto=off` will do.

File: .github/workflows/gh-actions.yml [hunk 3]
```diff
@@ -129,6 +129,18 @@ jobs:
...
+    runs-on: ubuntu-latest
+    env:
+      CC: clang
+    steps:
+      - uses: actions/checkout@v2
+      - run: sudo apt update
+      - run: sudo apt install gcc-multilib
+      - run: $RUNNER -q --unittests
+      # clang has bug in supporting lto 
+      - run: $RUNNER -q --buildoption-test --buildoptions=--lto=off,--jerry-math=on
```
Comment (ossy-szeged) [inline .github/workflows/gh-actions.yml:142]: Why do we need --jerry-math=on for buildoption tests?

Comment (gengjiawen) [reply .github/workflows/gh-actions.yml]: Previously one test case will failed, maybe a cache issue. I will give it another try.

**Review by ossy-szeged — APPROVED** [review]
LGTM

**Review by rerobika — APPROVED** [review]

**Pull request merged**